docs/07_how-to/01_auth.mdx

```diff
@@ -0,0 +1,395 @@
+
+[comment]: <> (Describe user interface to assign permission codes and users to profiles.)
+[comment]: <> (Show a video of adding users? Also add some screen shots?)
+
+
+## Code examples
+
+:::info Take a look at the code
+Check our [example application](https://github.com/genesislcap/platform-howto/tree/master/howto-permissions), which shows the permission features that are available.
+
+:::
+
+The example application has a single data table called Trade.  After a remap, this will be populated with 10 entries - 5 in each of 2 Countries.  Two further tables are added:
+
+- Countries (single column)
+- Mapping (which maps between User and Country to enable row authorisation based upon permitted Countries)
+
+The front-end grid demonstrates different authorisations in each quadrant - these map directly to four different Data Server queries, which you can see in the file **Permissions-dataserver.kts**.
+
+Only Users assigned to the Profile **Full_Access** can insert or modify trades. Currently, this is the user **AmyAccess**, but more Profiles and Users can be added to the application once it is installed and running.
+
+For example, you can edit the User Country table on the **Static** tab to remove Canada from AmyAccess. Once you have done that, AmyAccess will only be able to enter trades against the UK. You can see the code in the file **Permissions-eventhandler.kts**.
+
+Note that only the user **admin** can modify Profiles and add users. The password for all users who are set up in advance is **genesis**.
+
+The application also demonstrates authorisation at a dataserver and eventhandler level.
+
+:::info
+The **ReadMe** file in the project folder provides further information on the project.
+:::
+
+## A more complex scenario
+
+Consider how authorisation can manage more complex situations.
+
+For example, consider the lifecycle of a trade. Perhaps you want to ensure that only super users can modify trades that have settled. 
+
+To achieve this, we are going to modify an existing TRADE_MODIFY event and then create a new modify event for super users only.
+
+The following code modifies the earlier TRADE_MODIFY event to prevent users from changing the transaction if it has a status of `Settled`. This is defined in an [`onValidate`](/server/event-handler/basics/#adding-validation) codeblock:
+
+```jsx
+  eventHandler<Trade>("TRADE_MODIFY", transactional = true) {
+    permissioning {
+      permissionCodes = listOf("TradeUpdate")
+    }
+    onValidate { event ->
+        val trade = entityDb.get(Trade.ById(event.details.tradeId))!!
+        require(trade != null) { "Trade Id does not exist" }
+        require(trade.status != status.Settled) { "Trade is already settled" }
+        ack()
+    }
+    onCommit { event ->
+    ...
+    }
+  }
+```
+
+You can then easily create a separate event so that super users can modify any trade. In this event, you don't need to check the status - you simply make sure that it is only available to super users.
```

Comment:
@jacinpoz do you know if we have access to permissionsCodes a user has in onValidate easily? Creating a new event feels OTT compared to just putting in onValidate block (+ permissionCodes list) if you know what I mean.

Not a blocker for this release but wondering if we're misleading people here